lib/connections/limiter.go

```diff
@@ -124,6 +127,87 @@ func (lim *limiter) VerifyConfiguration(from, to config.Configuration) error {
 	return nil
 }
 
+func nextChange(to config.Configuration) time.Duration {
+	now := time.Now()
+
+	beginHour := to.Options.ScheduledRatesFromHour
+	beginMinute := to.Options.ScheduledRatesFromMinute
+	begin := time.Date(now.Year(), now.Month(), now.Day(), beginHour, beginMinute, 0, 0, now.Location())
+
+	tillBegin := begin.Sub(now)
```

Comment:
This entire time dance happens a few times, worth making a small utility function like e.g. `timeDiff(now time.Time, day, hour, minute int) time.Duration`

Reply:
Done